### ISSUE TITLE
[XMPP] contacts are never marked as online (breaks OTR too!)

### ISSUE THREAD

**Issue by acidicX** [loqui/im#683]
With loqui v0.4, all my XMPP contacts are always offline on app start (even if they are not). The online state does not even change when they are sending me messages or OTR becomes active. They only appear online if they reconnect their XMPP account while my loqui app is online.

I did not have this problem with 0.3.4.

**Comment by acidicX** [comment]
With 0.4.1a und 0.4, they will not show online anymore even if the contacts reconnect after loqui is online. This also means that OTR cannot be initialized!

**Comment by TitanNano** [comment]
Thank you for the report. I will take a look at it as soon as possible!

**Comment by TitanNano** [comment]
accounts should show online again.

**Comment by acidicX** [comment]
I just pushed the newest git to my device, contacts are still offline - they only appear online when they reconnect after loqui is online :(

**Comment by acidicX** [comment]
I just noticed that OTR works again now, even if the contacts are marked offline. So loqui must recognize that the contact is online..? Maybe that helps..

**Comment by TitanNano** [comment]
I did some tests, I can not confirm this. I used two instances of loqui and as soon as the response for the disco#info request arrives the  contact is marked as away or available.

**Comment by acidicX** [comment]
Not for me. Are you on the 3.0 nightly as well? Maybe this is a bug in gecko/gaia - but the communication layer works, so it is more likely an issue with loqui.
I haven't checked this on a 2.1 device yet, but when I first reported the bug it appeared there as well. I can see from the console that loqui receives disco#info events. This is the console output for the jabber client I am looking for:

```
<iq from='account@jabberserver.xyz/19760469261433753984639373' to='myaccount@jabberserver.xyz/Loqui' type='get' id='purplede90a4a1' xmlns='jabber:client' xmlns:stream='http://etherx.jabber.org/streams' version='1.0'><query xmlns='http://jabber.org/protocol/disco#info' node='http://loqui.im#rh7a1AdpdrMB9n4CAlZwADn1nws=='/></iq>
```

**Comment by TitanNano** [comment]
I'm on v2.2.

I see and where is the disco#info response?

2015-06-08 13:44 GMT+02:00 Carsten notifications@github.com:

**Comment by acidicX** [comment]
This is from the response? (the console output was prefixed "RECV" - so it should be the response?)
Here is the entry prefixed by "SENT":

```
<iq type='result' id='purplede90a603' xmlns='jabber:client' to='account@jabberserver.xyz/19760469261433753984639373'><query xmlns='http://jabber.org/protocol/disco#info' node='http://loqui.im#rh7a1AdpdrMB9n4CAlZwADn1nws=='><identity category='client' type='phone' name='http://loqui.im'/><feature var='http://jabber.org/protocol/caps'/><feature var='http://jabber.org/protocol/disco#info'/><feature var='http://jabber.org/protocol/disco#items'/><feature var='urn:xmpp:attention:0'/><feature var='http://jabber.org/protocol/chatstates'/><feature var='urn:xmpp:delay'/><feature var='urn:xmpp:time'/><feature var='vcard-temp'/><feature var='urn:xmpp:receipts'/></query></iq>
```

**Comment by TitanNano** [comment]
RECV just means that is was an incoming message. The message you posted
first actually was a disco#info request form one of your contacts. The
second message is your response to a disco#info request.

2015-06-08 18:18 GMT+02:00 Carsten notifications@github.com:

**Comment by acidicX** [comment]
Ah ok. So what does a response look like? Could you give me a template or keyword so I can filter the console output for the disco#info response? I don't really know what to look for... Thanks!

**Comment by TitanNano** [comment]
Well your disco#info request should look like the one your got, but sent form Loqui, not received.

```
<iq from='your account here' to='your contacts account here' type='get' id='some random id' xmlns='jabber:client' xmlns:stream='http://etherx.jabber.org/streams' version='1.0'><query xmlns='http://jabber.org/protocol/disco#info' node='http://loqui.im#rh7a1AdpdrMB9n4CAlZwADn1nws=='/></iq>
```

For the response you can just search for the Id of your request because the response should have the same id.

**Comment by acidicX** [comment]
Ok, well, there are none. Loqui just answers incoming requests from clients, but never requests disco#info itself. The only two logs including

```
to='your contacts account here'
```

are "SENT" entries and answers to previous requests from my contacts account.

// edit:
If the other client (contact account) reconnects, loqui requests disco#info fine and also displays it correctly after getting a response.

```
"SENT" ... <iq type='get' to='contactaccount' from='myaccount/Loqui' xmlns='jabber:client' id='7984:sendIQ'><vCard xmlns='vcard-temp'/></iq><iq from='myaccount/Loqui' to='contactsaccount' type='get' xmlns='jabber:client' id='7985:sendIQ'><query xmlns='http://jabber.org/protocol/disco#info' node='http://pidgin.im/#I22W7CegORwdbnu0ZiQwGpxr0Go='/></iq><iq type='get' to='contactaccount' from='myaccount/Loqui' xmlns='jabber:client' id='7986:sendIQ'><vCard xmlns='vcard-temp'/></iq><iq from='myaccount/Loqui' to='contactaccount' type='get' xmlns='jabber:client' id='7987:sendIQ'><query xmlns='http://jabber.org/protocol/disco#info' node='http://pidgin.im/#I22W7CegORwdbnu0ZiQwGpxr0Go='/></iq>
```

**Comment by cmeerw** [comment]
I believe this is mostly fixed with commit 4c122afca3443f395cde2a98559e7febd5b3c3c5 (although there are still problem with multiple resources - looks like the strophe roster plugin needs some work)...
